libdnf/transaction/rpm_package.cpp

```diff
@@ -0,0 +1,199 @@
+            AND i.name = ?
+            AND i.arch = ?
+        ORDER BY
+            ti.trans_id DESC
+        LIMIT 1
+    )**";
+
+    if (arch != "") {
+        libdnf::utils::SQLite3::Query query(conn, sql);
+        query.bindv(name, arch);
+
+        if (query.step() == libdnf::utils::SQLite3::Statement::StepResult::ROW) {
+            auto action = static_cast<TransactionItemAction>(query.get<int64_t>("action"));
+            if (action == TransactionItemAction::REMOVE) {
+                return TransactionItemReason::UNKNOWN;
+            }
+            auto reason = static_cast<TransactionItemReason>(query.get<int64_t>("reason"));
+            return reason;
+        }
+    } else {
+        const char * arch_sql = R"**(
+            SELECT DISTINCT
+                arch
+            FROM
+                rpm
+            WHERE
+                name = ?
+        )**";
+
+        libdnf::utils::SQLite3::Query arch_query(conn, arch_sql);
+        arch_query.bindv(name);
+
+        TransactionItemReason result = TransactionItemReason::UNKNOWN;
+
+        while (arch_query.step() == libdnf::utils::SQLite3::Statement::StepResult::ROW) {
+            auto rpm_arch = arch_query.get<std::string>("arch");
+
+            libdnf::utils::SQLite3::Query query(conn, sql);
+            query.bindv(name, rpm_arch);
+            while (query.step() == libdnf::utils::SQLite3::Statement::StepResult::ROW) {
+                auto action = static_cast<TransactionItemAction>(query.get<int64_t>("action"));
+                if (action == TransactionItemAction::REMOVE) {
+                    continue;
+                }
+                auto reason = static_cast<TransactionItemReason>(query.get<int64_t>("reason"));
+                if (reason > result) {
+                    result = reason;
+                }
+            }
+        }
+        return result;
+    }
+    return TransactionItemReason::UNKNOWN;
+}
+*/
+
+/**
+ * Compare RPM packages
+ * This method doesn't care about compare package names
+ * \param other Package to compare with
+ * \return true if other package is newer (has higher version and/or epoch)
+ */
+bool Package::operator<(const Package & other) const {
+    // compare epochs
+    int32_t epoch_diff = other.get_epoch_int() - get_epoch_int();
+    if (epoch_diff > 0) {
+        return true;
+    } else if (epoch_diff < 0) {
+        return false;
+    }
+
+    // compare versions
+    std::stringstream versionThis(get_version());
+    std::stringstream versionOther(other.get_version());
+
+    std::string bufferThis;
+    std::string bufferOther;
+    while (std::getline(versionThis, bufferThis, '.') && std::getline(versionOther, bufferOther, '.')) {
+        int subVersionThis = std::stoi(bufferThis);
+        int subVersionOther = std::stoi(bufferOther);
+        if (subVersionThis == subVersionOther) {
+            continue;
+        }
+        return subVersionOther > subVersionThis;
+    }
+    return false;
+}
+
+/*
```

Comment:
Please could you add a TODO with description of future plans?

Reply:
I've added the TODO in the header file already.